### PR TITLE
Integration test for permissions changes

### PR DESCRIPTION
**Pull request by meain**
This adds basic integration tests for backing up permissions in OneDrive.
*https://github.com/alcionai/corso/issues/2790 could be a good follow up to this.*

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [x] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* fixes https://github.com/alcionai/corso/issues/2772

#### Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/operations/backup_integration_test.go [hunk 1]
```diff
@@ -455,6 +455,29 @@ func toDataLayerPath(
...
+	d, err := service.Client().UsersById(userID).Drive().Get(ctx, nil)
+	if err != nil {
+		err = graph.Wrap(
+			ctx,
+			err,
+			"retrieving default user drive").
+			With("user", userID)
+	}
+
+	require.NoError(t, err)
```
Comment (ryanfkeepers) [inline src/internal/operations/backup_integration_test.go:473]: Nit: add the clues error core for debugging
```suggestion
	require.NoError(t, err, clues.ToCore(err))
```

File: src/internal/operations/backup_integration_test.go [hunk 2]
```diff
@@ -1254,6 +1252,102 @@ func (suite *BackupOpIntegrationSuite) TestBackup_Run_oneDriveIncrementals() {
...
+						SharingMode: onedrive.SharingModeCustom,
+						Permissions: []onedrive.UserPermission{
+							{
+								Roles:    []string{"write"},
+								EntityID: suite.user,
+							},
+						},
+					},
+				)
+				require.NoError(t, err, "add permission to file", clues.ToCore(err))
```
Comment (ryanfkeepers) [inline src/internal/operations/backup_integration_test.go:1276]: Using core with a string will require a formatting target.
```suggestion
				require.NoError(t, err, "add permission to file %v", clues.ToCore(err))
```

**Review by ryanfkeepers — APPROVED** [review]

**Review by ashmrtn — COMMENTED** [review]
more a high-level question, but if we're not passing permission maps around then how are we merging the permission set to figure out what needs added/removed? Are we going to rely on fetching the permissions from Graph inline?

**Comment by meain** [comment]
> more a high-level question, but if we're not passing permission maps around then how are we merging the permission set to figure out what needs added/removed? Are we going to rely on fetching the permissions from Graph inline?

That is what we are doing currently, might have rework that soon ish.

**Pull request merged**